lms/validation/blackboard.py

```diff
@@ -0,0 +1,26 @@
+"""
```

Comment:
I don't think this file is in the right place. `lms/validation` is for schema applied before views. This should be next to (or ideally in) the code it's used in.

Reply:
Done.

I don't agree that it's ideal for the Blackboard API client's schemas to be _in_ the `BlackboardAPIClient` class itself or in the same file for reasons I discussed above: I think this makes the Canvas API client confusing and hard to read because the class/file is too long and because it contains a long mixture of two different types of thing. It won't scale to keep adding methods and schemas to one class/file, it'll get too long. I also think the integrated tests are problematic (also discussed in the other comments). I think schemas are better treated as separate units.

So I ended up moving the service to `services/blackboard_api/service.py` and the schemas to `services/blackboard_api/_schemas.py` where it's clear that they're private to the `services/blackboard_api/` directory.

I think there might be a best-of-both-worlds here where we put the `list_files()` method and `ListFilesSchema` in one file, the `public_url()` method and `PublicURLSchema` in another file, and so no. Allowing each method + schema to be next to each other in the same file but avoiding a really long file that's hard to read. But I'm not sure exactly how that would look or whether it would actually turn out to be an improvement in the end. I think the existing layout is serviceable enough so for practicality (we are on a tight deadline here) I'm gonna vote for not letting the perfect be the enemy of good and going with this for now.